django/db/backends/oracle/operations.py

```diff
@@ -53,6 +54,16 @@ class DatabaseOperations(BaseDatabaseOperations):
     # Oracle doesn't support string without precision; use the max string size.
     cast_char_field_without_max_length = 'NVARCHAR2(2000)'
 
+    unsupported_distinct_aggregates = (
+        aggregates.StdDev,
+    )
+
+    def check_expression_support(self, expression):
+        if isinstance(expression, self.unsupported_distinct_aggregates) and expression.distinct:
+            raise NotImplementedError(
```

Comment:
`NotSupportedError` per PEP 249.

Reply:
Makes sense, it might be worth auditing our uses of `NotImplementedError` as there's a few instances that should be changed. Do you want to tackle it? Else I'll file a ticket and add it to this PR.

Reply:
Handled in #9186.